### PR TITLE
sdk: Rework credentials apis

### LINKED ISSUE #469: sdk: Instead of having Credentials.CreateForXXX functions, use enums
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG

**What happened**:
As pointed to @disrani-px we need to change the credentials CreateForXXX and EnumerateForXXX and have them take an enum of the cloud provider as one of the values in the request.

### PR DESCRIPTION
**Pull request by lpabon**
**What this PR does / why we need it**:
Instead of having an API per credential type, we now use the OneOf tags in proto files, making it much easier to use and implement.

**Which issue(s) this PR fixes** (optional)
Closes #469 
Part of #407 

**Special notes for your reviewer**:
* Added Inspect call.
* Generated files moved to their own commit.

### CODE REVIEW

File: api/server/sdk/credentials_test.go [hunk 1]
```diff
@@ -598,90 +620,11 @@ func TestSdkCredentialEnumerateAzureSuccess(t *testing.T) {
 		// Setup client
 	c := api.NewOpenStorageCredentialsClient(s.Conn())
 
-	// Enumerate Azure Credentials
-	resp, err := c.EnumerateForAzure(context.Background(), req)
+	// Inspect
+	resp, err := c.Inspect(context.Background(), req)
```
Comment (ram-infrac) [inline api/server/sdk/credentials_test.go]: can you please add example of how to use Inspect resp here?

Comment (lpabon) [reply api/server/sdk/credentials_test.go]: Yeah, great idea.

File: api/server/sdk/credentials_test.go [hunk 2]
```diff
@@ -241,19 +249,19 @@ func TestSdkAzureCredentialCreateBadArgument(t *testing.T) {
...
 	defer s.Stop()
 
-	req := &api.SdkCredentialCreateAzureRequest{}
-
-	params := make(map[string]string)
-
-	params[api.OptCredType] = "azure"
-	params[api.OptCredAzureAccountKey] = req.GetCredential().GetAccountKey()
-	params[api.OptCredAzureAccountName] = req.GetCredential().GetAccountName()
+	req := &api.SdkCredentialCreateRequest{
```
[the remark this answers is not in the document]
Comment (lpabon) [reply api/server/sdk/credentials_test.go]: Doesn't look like an issue to me.

File: api/server/sdk/credentials_test.go (hunk 1 above)
Comment (lpabon) [reply api/server/sdk/credentials_test.go]: Actually, there is an example on line 627, but I'll add more info to the proto file docs.

**Pull request merged**